src/coreclr/src/vm/dwbucketmanager.hpp

```diff
@@ -1215,11 +1215,14 @@ void CLR20r3BucketParamsManager::PopulateBucketParameters()
     CONTRACTL
     {
         NOTHROW;
-        GC_NOTRIGGER;
+        GC_TRIGGERS;
         MODE_ANY;
     }
     CONTRACTL_END;
 
+    // Preempt to let GC suspend
+    GCX_PREEMP();
+
```

Comment:
does `PREEMP` require a revert or will the thread switch to `GCX_COOP` when required?

Reply:
`GCX_PREEMP` does not require revert. The macro expands into an instance of an object with destructor. The object destructor will take care of the reverting at the end of the scope.